tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/AmqpLargeMessageTest.java

```diff
@@ -265,6 +266,51 @@ public void testSendAMQPReceiveAMQPViaJMSBytes() throws Exception {
       return payload;
    }
 
+   @Test(timeout = 60000)
+   public void testSendHugeHeader() throws Exception {
+      doTestSendHugeHeader(PAYLOAD);
+   }
+
+   @Test(timeout = 60000)
+   public void testSendLargeMessageWithHugeHeader() throws Exception {
+      doTestSendHugeHeader(1024 * 1024);
+   }
+
+   public void doTestSendHugeHeader(int expectedSize) throws Exception {
+      server.getAddressSettingsRepository().addMatch("#", new AddressSettings().setDefaultAddressRoutingType(RoutingType.ANYCAST));
+
+      AmqpClient client = createAmqpClient();
+      AmqpConnection connection = addConnection(client.connect());
+      try {
+
+         connection.connect();
+
+         final int strLength = 1024 * 1024;
+         AmqpSession session = connection.createSession();
+         AmqpSender sender = session.createSender(testQueueName);
+
+         AmqpMessage message = createAmqpMessage((byte) 'A', expectedSize);
+         StringBuffer buffer = new StringBuffer();
+         for (int i = 0; i < strLength; i++) {
+            buffer.append(" ");
+         }
+         message.setApplicationProperty("str", buffer.toString());
+         message.setDurable(true);
+
+         try {
+            sender.send(message);
+            fail();
+         } catch (IOException e) {
+            Assert.assertTrue(e.getCause() instanceof JMSException);
+            Assert.assertTrue(e.getMessage().contains("larger than the store maxRecordSize"));
```

Comment:
Please, assert for the Exception Code here.

on this case... contains("AMQ149005")

you will understand why based on my next comment

Reply:
I changed the check to test Exception Code.